assets/training/oai_completions_finetune/spec.yaml

```diff
@@ -0,0 +1,105 @@
+    type: string
+    optional: False
+    description: User-defined registered model name
+  train_dataset:
+    type: uri_folder
+    optional: False
+    description: Input dataset (file or folder). If a folder dataset is passed, includes all nested files.
+  validation_dataset:
+    type: uri_folder
+    optional: True
+    description: Input dataset (file or folder). If a folder dataset is passed, includes all nested files.
+  lora_weights:
+    type: uri_folder
+    description: LoRA weights for continual finetuning. This is optional.
+    optional: True
+  n_epochs:
+    type: integer
+    optional: True
+    default: 4
+    description: Number of epochs for the training
+  batch_size:
+    type: integer
+    optional: True
+    default: -1
+    description: The batch size to use for training. When set to -1, batch_size is calculated as 0.2% of examples in training set and the max is 256.
+  learning_rate_multiplier:
+    type: number
+    optional: True
+    default: 0.1
+    description: The learning rate multiplier to use for training. Must be between 0.0 and 5.0.
+  prompt_loss_weight:
+    type: number
+    optional: True
+    default: 0.1
+    min: 0
+    max: 1
+    description: The prompt loss weight to use for training
+  compute_classification_metrics:
+    type: boolean
+    optional: True
+    description: If set, we calculate classification-specific metrics such as accuracy and F-1 score using the validation set at the end of every epoch. In order to compute classification metrics, you must provide a validation_file. Additionally, you must specify classification_n_classes for multiclass classification or classification_positive_class for binary classification.
+  classification_n_classes:
+    type: integer
+    optional: True
+    description: The number of classes in a classification task. This parameter is required for multiclass classification.
+  classification_positive_class:
+    type: string
+    optional: True
+    description: The positive class in binary classification. This parameter is needed to generate precision, recall, and F1 metrics when doing binary classification.
+  classification_betas:
+    type: string
+    optional: True
+    description: If this is provided, we calculate F-beta scores at the specified beta values. The F-beta score is a generalization of F-1 score. This is only used for binary classification. With a beta of 1 (i.e. the F-1 score), precision and recall are given the same weight. A larger beta score puts more weight on recall and less on precision. A smaller beta score puts more weight on precision and less on recall. The value specified should be a comma separated list of doubles.
+
+outputs:
+  output_model:
+    type: uri_folder
+    mode: mount
+    description: Dataset with the output model weights (LoRA weights)
+
+jobs:
+  fine_tune:
+    type: command
+    component: azureml://registries/azure-openai-preview/components/openai_completions_finetune/versions/0.3.0
```

Comment:
should this be in azureml registry?

Reply:
It cant. This component requires IPP (Intellectual Property Protection) registry, azureml is not the one. Currently registry can only be either one IPP or non-IPP. There is a work going to make registry support both but its not there yet.

Reply:
do you want to publish the asset in IPP registry then?